src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -315,70 +313,44 @@ private static string GenerateForTypeWithKnownConverter(TypeGenerationSpec typeM
                 string typeCompilableName = typeMetadata.TypeRef;
                 string typeFriendlyName = typeMetadata.TypeInfoPropertyName;
 
-                string metadataInitSource = $@"_{typeFriendlyName} = {JsonMetadataServicesTypeRef}.{GetCreateValueInfoMethodRef(typeCompilableName)}({OptionsInstanceVariableName}, {JsonMetadataServicesTypeRef}.{typeFriendlyName}Converter);";
+                string metadataInitSource = $@"{CreateTypeInfoReturnValueLocalVariableName} = {JsonMetadataServicesTypeRef}.{GetCreateValueInfoMethodRef(typeCompilableName)}({OptionsLocalVariableName}, {JsonMetadataServicesTypeRef}.{typeFriendlyName}Converter);";
 
                 return GenerateForType(typeMetadata, metadataInitSource);
             }
 
             private static string GenerateForTypeWithUnknownConverter(TypeGenerationSpec typeMetadata)
             {
                 string typeCompilableName = typeMetadata.TypeRef;
-                string typeFriendlyName = typeMetadata.TypeInfoPropertyName;
 
                 // TODO (https://github.com/dotnet/runtime/issues/52218): consider moving this verification source to common helper.
                 StringBuilder metadataInitSource = new(
                     $@"{JsonConverterTypeRef} converter = {typeMetadata.ConverterInstantiationLogic};
-                    {TypeTypeRef} typeToConvert = typeof({typeCompilableName});");
+                {TypeTypeRef} typeToConvert = typeof({typeCompilableName});");
 
-                if (typeMetadata.IsValueType)
-                {
-                    metadataInitSource.Append($@"
-                        if (!converter.CanConvert(typeToConvert))
-                        {{
-                            {TypeTypeRef}? underlyingType = {NullableTypeRef}.GetUnderlyingType(typeToConvert);
-                            if (underlyingType != null && converter.CanConvert(underlyingType))
-                            {{
-                                // Allow nullable handling to forward to the underlying type's converter.
```

Comment:
This logic was meant to match converters to nullable properties. How is this fallback handled currently?

Reply:
This is what we've discussed offline, this code is wrong and we will never get into that branch.

Reply:
I'll refrain from fixing this now. I prefer to get the feature in and then do bug fixes separately